packages/java-edition/src/mcfunction/parser/argument.ts

```diff
@@ -189,7 +190,7 @@ export const argument: mcf.ArgumentParserGetter = (
 		case 'minecraft:item_predicate':
 			return wrap(itemPredicate)
 		case 'minecraft:item_slot':
-			return wrap(core.literal(...ItemSlotArgumentValues))
+			return wrap(core.literal(...getItemSlotArgumentValues(ctx)))
```

Comment:
This is how you could get the `ctx` for the item slots without requiring the entire `argument` function to need the context.
```suggestion
			return wrap((src, ctx) => {
			  return core.literal(...getItemSlotArgumentValues(ctx))(src, ctx)
			})
```

Reply:
I'm having trouble with grokking this intuitively.

Reply:
`core.literal()` is a function that returns a `Parser` (a `Parser` is (basically) a function in the form of `(src, ctx) -> AstNode`). It is functionally equivalent to write:

```ts
core.literal(...)
```

and 

```ts
(src, ctx) => core.literal(...)(src, ctx)
```

as the first one returns a `Parser` directly, and in the second one we are writing a parser (the `(src, ctx) => ...` part) that returns the result of calling the `core.literal(...)` parser with `(src, ctx)`. Misode's suggestion turns the original form into the second form so `getItemSlotArgumentValues()` has access to `ctx` from the `Parser` being called.